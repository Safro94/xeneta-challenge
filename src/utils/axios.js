import axios from 'axios';
import https from 'https';

const opts = {
	httpsAgent: new https.Agent({ rejectUnauthorized: false }),
	baseURL: process.env.REACT_APP_SERVER_URL,
};

const interceptorError = error => Promise.reject(error.response);

const instance = axios.create(opts);

instance.interceptors.response.use(response => response, interceptorError);
instance.defaults.headers.common['x-api-key'] = process.env.REACT_APP_API_KEY;

export default instance;
