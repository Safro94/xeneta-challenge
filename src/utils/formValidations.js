const validateSearchForm = ({
	departure,
	destination,
	departureDate,
	returnDate,
}) => {
	if (!departure || !destination || !departureDate) {
		return false;
	}

	if (departure.code === destination.code) {
		return false;
	}

	if (returnDate && returnDate < departureDate) {
		return false;
	}

	return true;
};

const errorCompare = (trip, message) => {
	const { departure, destination } = trip;

	const status = !departure && !destination ? true : false;

	return {
		message,
		status: !status && departure?.code === destination?.code,
	};
};

export { validateSearchForm, errorCompare };
