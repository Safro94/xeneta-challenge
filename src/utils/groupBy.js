import { marketPositions } from 'constants/marketPositions';

const groupBy = data =>
	data.reduce((accum, item) => {
		if (!accum[marketPositions.low]) accum[marketPositions.low] = [];
		if (!accum[marketPositions.average]) accum[marketPositions.average] = [];
		if (!accum[marketPositions.high]) accum[marketPositions.high] = [];

		accum[marketPositions.low].push({ x: item.day, y: item.low });
		accum[marketPositions.average].push({ x: item.day, y: item.mean });
		accum[marketPositions.high].push({ x: item.day, y: item.high });

		return accum;
	}, {});

export default groupBy;
