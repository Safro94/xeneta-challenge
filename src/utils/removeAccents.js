const removeAccents = value => {
	const map = {
		a: 'á|à|ã|â',
		e: 'é|è|ê',
		i: 'í|ì|î',
		o: 'ó|ò|ô|õ',
		u: 'ú|ù|û|ü',
	};

	for (let pattern in map) {
		value = value.replace(new RegExp(map[pattern], 'g'), pattern);
	}

	return value;
};

export { removeAccents };
