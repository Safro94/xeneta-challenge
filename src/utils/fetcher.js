import axios from 'utils/axios';

const fetcher = async ({ method = 'get', url, data = null }) => {
	return axios[method](url, JSON.parse(data)).then(
		response => response.data,
		error => {
			throw error;
		}
	);
};

export default fetcher;
