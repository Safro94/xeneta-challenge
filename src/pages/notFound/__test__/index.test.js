import { render } from 'utils/test-utils';
import NotFound from '../';

describe('Not Found', () => {
	it('should match the snapshot', () => {
		const { container } = render(<NotFound />);
		expect(container.firstElementChild).toMatchSnapshot();
	});
});
