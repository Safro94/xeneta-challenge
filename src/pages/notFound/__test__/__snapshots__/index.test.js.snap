// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found should match the snapshot 1`] = `
<div
  class="container"
>
  <i
    class="icon"
  >
    <svg
      class="icon"
      viewBox="0 0 130 167"
    >
      <defs>
        <style>
          .prefix__a{fill:#36c7d6;opacity:.15}.prefix__e{fill:#fff}.prefix__c,.prefix__e{stroke:#36c7d6}.prefix__d{stroke-linejoin:round}.prefix__c,.prefix__d,.prefix__e{stroke-width:2px}.prefix__c{fill:#9ce3eb}.prefix__c,.prefix__e{stroke-miterlimit:10}.prefix__d{fill:none;stroke-linecap:round;stroke:#fff}
        </style>
      </defs>
      <title>
        404-icon-8
      </title>
      <ellipse
        class="prefix__a"
        cx="70"
        cy="162"
        rx="60"
        ry="5"
      />
      <path
        d="M71.71 1H30.29L1 30.29v41.42L30.29 101h41.42L101 71.71V30.29L71.71 1z"
        fill="#fff"
        stroke="#36c7d6"
        stroke-linejoin="round"
        stroke-width="2"
      />
      <path
        class="prefix__c"
        d="M69.64 6H32.36L6 32.36v37.28L32.36 96h37.28L96 69.64V32.36L69.64 6z"
      />
      <path
        class="prefix__d"
        d="M34.15 63.87V39.22L19.32 56.78h17.92M77.23 63.87V39.22L62.4 56.78h17.92"
      />
      <ellipse
        class="prefix__d"
        cx="50.34"
        cy="51.55"
        rx="9.13"
        ry="12.32"
      />
      <path
        class="prefix__e"
        d="M46 101h10v60H46z"
      />
      <path
        class="prefix__a"
        d="M46 101h10v6H46z"
      />
      <path
        class="prefix__e"
        d="M101 161H53l16-70h16l16 70z"
      />
      <path
        d="M37 161h80"
        fill="none"
        stroke="#36c7d6"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
      <path
        class="prefix__c"
        d="M63.29 116h27.42l-1.82-8H65.11l-1.82 8zM59.17 134h35.66L93 126H61l-1.83 8z"
      />
    </svg>
  </i>
  <h2>
    Go back to the
     
    <a
      class="link"
      href="/"
    >
      home page
    </a>
  </h2>
</div>
`;
